caffe2/operators/boolean_mask_ops.cc

```diff
@@ -304,8 +304,8 @@ void MaskWithFunctor(
     // M is product of dimensions at/after axis
     // then mask each batch [i, :, :] (note that this is N x M matrix)
     for (int i = 0; i < B; ++i) {
-      for (int j = 0; j < N; ++j) {
-        for (int k = 0; k < M; ++k) {
+      for (size_t j = 0; j < N; ++j) {
```

Comment:
This also changes the size to 64-bit. We should use uint32_t here, otherwise there might be performance differences.

Reply:
Actually, in  this case, it looks more reasonable to change `M` and `N` types from `size_t` to `int`